### PR TITLE
Udpate LRO pattern to use new SLRO

### PR DESCRIPTION
**Pull request by ShivangiReja**
# Description
Issue: https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/6103

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: samples/Azure.Management.Storage/Generated/BlobContainer.cs [hunk 1]
```diff
@@ -182,15 +150,19 @@ public async virtual Task<BlobContainerDeleteOperation> StartDeleteAsync(Cancell
         }
 
         /// <summary> Deletes specified container under its account. </summary>
+        /// <param name="waitForCompletion"> Waits for the completion of the long running operations. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual BlobContainerDeleteOperation StartDelete(CancellationToken cancellationToken = default)
+        public virtual BlobContainerDeleteOperation Delete(bool waitForCompletion = true, CancellationToken cancellationToken = default)
```
Comment (allenjzhang) [inline samples/Azure.Management.Storage/Generated/BlobContainer.cs:155]: >Delete [](http://example.com/codeflow?start=52&length=6)

Response<BlobContainerDeleteOperation>?

**Comment by allenjzhang** [comment]
>                 WriteClientMethod(_writer, restMethod, restMethod.Name, _resourceContainer.GetDiagnostic(restMethod.RestClientMethod), _resourceContainer.OperationGroup, _context, true);

is LRO respected for RemainingMethods?

---
Refers to: src/AutoRest.CSharp/Mgmt/Generation/ResourceContainerWriter.cs:104 in 1a7744b. [](commit_id = 1a7744b9ae4ab93f8cee160d385fe8732ec6e437, deletion_comment = False)

**Review by ShivangiReja — COMMENTED** [review]
Test

File: src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs [hunk 2]
```diff
@@ -441,6 +441,11 @@ public IEnumerable<MgmtNonResourceOperation> GetNonResourceOperations(string par
             return _childResources;
         }
 
+        internal bool IsLongRunningReallyLong(RestClientMethod clientMethod)
+        {
+            return clientMethod.Operation.IsLongRunningReallyLong != null ? clientMethod.Operation.IsLongRunningReallyLong.Value : false;
```
Comment (allenjzhang) [inline src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs:446]: nit: return clientMethod.Operation.IsLongRunningReallyLong??false;

**Comment by allenjzhang** [comment]
>             "post": {

Can we add a S/LRO post operation? One real example is VM poweroff.

---
Refers to: test/TestProjects/MgmtLRO/mgmtLRO.json:284 in 1a7744b. [](commit_id = 1a7744b9ae4ab93f8cee160d385fe8732ec6e437, deletion_comment = False)

**Review by allenjzhang — CHANGES_REQUESTED** [review]
Once we have examples of LRO and SLRO for post operations, I am good to merge the PR.

**Review by allenjzhang — APPROVED** [review]

**Pull request merged**